src/structures/AutocompleteInteraction.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+const BaseCommandInteraction = require('./BaseCommandInteraction');
+const CommandInteractionOptionResolver = require('./CommandInteractionOptionResolver');
+const { InteractionResponseTypes } = require('../util/Constants');
+
+/**
+ * Represents an autocomplete interaction.
+ * @extends {BaseCommandInteraction}
+ */
+class AutocompleteInteraction extends BaseCommandInteraction {
+  constructor(client, data) {
+    super(client, data);
+
+    /**
+     * The options passed to the command
+     * @type {CommandInteractionOptionResolver}
+     */
+    this.options = new CommandInteractionOptionResolver(
+      this.client,
+      data.data.options?.map(option => this.transformOption(option, data.data.resolved)) ?? [],
+      this.transformResolved(data.data.resolved ?? {}),
+    );
+  }
+
+  /**
+   * Sends results for the autocomplete of this interaction.
+   * @param {InteractionResultOptions} options The options for the autocomplete
```

Comment:
`InteractionResultOptions` needs defining!

Reply:
Could you clarify what you mean? Cause the type can be found on line 4087 in index.d.ts so I'm unsure of what you mean to define it.

Reply:
This is JSDoc. You've used it without creating a `@typedef` for it, so it won't show up in the documentation. If you search around the repository, you'll see every custom type has an associated `@typedef` for it (and those that don't won't show up on the documentation). All you have to do is create one and you're good to go.

This will actually be the same as [`ApplicationCommandOptionChoice`](https://discord.js.org/#/docs/main/main/typedef/ApplicationCommandOptionChoice), so lucky for you it's just a copy-paste.

Reply:
nevermind, my bad.